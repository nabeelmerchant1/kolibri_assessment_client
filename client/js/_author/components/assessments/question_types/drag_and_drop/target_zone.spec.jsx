import React          from 'react';
import { shallow }    from 'enzyme';
import TargetZone     from './target_zone';

describe('target_zone component', () => {
  let props;
  let result;

  beforeEach(() => {
    props = {
      target: {},
      zones: {},
      editZone: () => {},
      openModal: () => {},
    };
    result = shallow(<TargetZone {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles setting state', () => {
    expect(result.instance().state.activeZone).toBe(null);
    result.instance().state.activeZone = '7';
    expect(result.instance().state.activeZone).toBe('7');
    result.find('.au-c-drag-and-drop__target-image').simulate('click');
    expect(result.instance().state.activeZone).toBe(null);
  });
});
