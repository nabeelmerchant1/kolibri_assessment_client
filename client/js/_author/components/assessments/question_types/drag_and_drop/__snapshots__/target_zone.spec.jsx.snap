// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target_zone component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-c-drag-and-drop__target-image"
    onClick={[Function]}
>
    <targetPlaceholder
        onClick={[Function]}
    />
</div>,
  "nodes": Array [
    <div
      className="au-c-drag-and-drop__target-image"
      onClick={[Function]}
>
      <targetPlaceholder
            onClick={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TargetZone
        editZone={[Function]}
        openModal={[Function]}
        target={Object {}}
        zones={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TargetZone {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "editZone": [Function],
          "openModal": [Function],
          "target": Object {},
          "zones": Object {},
        },
        "refs": Object {},
        "state": Object {
          "activeZone": null,
        },
        "target": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-c-drag-and-drop__target-image"
          onClick={[Function]}
>
          <targetPlaceholder
                    onClick={[Function]}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-c-drag-and-drop__target-image"
          onClick={[Function]}
>
          <targetPlaceholder
                    onClick={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TargetZone
    editZone={[Function]}
    openModal={[Function]}
    target={Object {}}
    zones={Object {}}
/>,
}
`;
